### ISSUE TITLE
Release ETA

### ISSUE THREAD

**Issue by jpike88** [sequelpro/sequelpro#3034]
I'm tracking issues many months old that are fixed... is there an ETA on when another version will take place?

**Comment by dmoagx** [comment]
Yeah, I'd like to know, too...

**Comment by jpike88** [comment]
Who is the guy we need to poke? Is someone the release manager?

**Comment by dmoagx** [comment]
@abhibeckert and @mattlangtree are usually doing the release stuff

**Comment by jpike88** [comment]
@abhibeckert, @mattlangtree if either of you are available, please can you answer, or point me to someone who can?

**Comment by jpike88** [comment]
@dmoagx you've been making commits exclusively over 6 months, and the actual guys responsible for releasing seem to be AWOL. What exactly is going on?

**Comment by dmoagx** [comment]
@jpike88 No idea, but the fact that we are [16.000 km apart from each other](https://www.youtube.com/watch?v=XfR9iY5y94s) may contribute to that

**Comment by jpike88** [comment]
Haha fair enough... looks like one of them has been active recently. @abhibeckert ... care to spare a minute for me? I mean... if the releasers are that hard to get a hold of, does that mean this project is indefinitely stalled, despite all of the commits by @dmoagx???

**Comment by jpike88** [comment]
@abhibeckert you made 20 contributions on June 5 (3 days ago), and still zero response here. Really?

**Comment by jpike88** [comment]
Looks like a fork is imminent, with its own name. If anyone has suggestions for a name now would be a good time to give them

**Comment by tschoffelen** [comment]
@jpike88  I'd love to help with that. I've recently registered the domain `prequel.app`, perhaps that's something we might use? Would love to continue this conversation with you elsewhere if you'd like.

**Comment by jpike88** [comment]
Yeah, sounds good to me! Anything we do I think should be laid out by @dmoagx I reckon

**Comment by dmoagx** [comment]
I think the main issue with getting that off the ground would be money.
We'd need a Mac Mini build server and Apple's current hardware lineup is craptacular. 4 year old hardware with the 16 GiB RAM that would be needed soldered in still costs 800€+. That plus the 100€/year for the (technically) worthless code signing certificate...
(And then there is all the small stuff like domains, web hosting, etc.)

**Comment by rowanbeentje** [comment]
I've no idea on how up to date the old nightly build script is, and I need to do some serious updating of build chains, but I still have an old mac mini that I can probably get to produce the nightlies again and upload them to somewhere. Release builds used to be done manually iirc!

**Comment by jpike88** [comment]
Considering the instability issues on macOS Mojave, I also suggest we mark this project as deprecated or sunset and redirect similarly confused users to the new one. As for website hosting etc etc, we could just leverage the GitHub pages functionality here, it supports HTTPS with custom domains now I believe.

**Comment by jpike88** [comment]
@dmoagx Is something like CircleCI adequate for builds? Is the build process long or intensive?

**Comment by abhibeckert** [comment]
Sorry I didn't see this thread in April for some reason, and didn't have time to reply yesterday when it first came to my attention. I don't have a whole lot of time right now either, writing this in the middle of the work day at my day job.

The current build system is available on a private github repository which is used for a variety of things that probably shouldn't be available for the whole world to see. In essence it works like this:

* the sequelpro.com server watches for new code on github, and creates an entry in it's own database which can be viewed at sequelpro.com/test-builds
* A simple mac app, polls for "pending" builds on that list, and runs a shell script to pull down the code, perform a build and upload to sequelpro.com which will make it available via Sparkle. Ideally this would run on multiple macs, but currently it's only on a Mac Mini that's in my home office
* the build script is somewhat complicated because of localization and code signing, one of those two is probably the cause of it not working right now, though I haven't had time to look into it.

jpike88 if you'd like to troubleshoot I'm happy to give you access to the code. dmoagx and rowanbeentje should already have access to it, as they wrote as much of the code as I did.

As for when the next release will be, I'd like to do one soon — during the macOS 10.14 developer preview/beta test period. Here the list of issues planned for the release:

https://github.com/sequelpro/sequelpro/projects/2

I haven't added #3074 to that list yet, and there are probably a few more.

**Comment by jpike88** [comment]
Thank the gods, we were all really worried that the wheels fell off!

Am happy to help however I can.

**Comment by abhibeckert** [comment]
We have a slack channel if you'd like to get involved? It's a bit easier to talk there, especially because of timezones/etc.

I'd send you a slack invite but I don't see a public email address on your github profile... can you send it to me? You can email it to contact@sequelpro.com if you'd prefer not to post it here.

**Comment by tschoffelen** [comment]
Would lobe to be involved as well if possible: thomas at includable dot com.

**Comment by dmoagx** [comment]
>  because of localization and code signing, one of those two is probably the cause of it not working right now

That would be plausible, since the broken crash reports server also hosts the translation site.
Anyway that needs to be fixed, too, since I don't want to provide new nightly builds with lots of changes without the crash reporter working.

**Comment by abhibeckert** [comment]
The server that translate.sequelpro.com is down, which would explain why builds aren't working... I'll look into it on the weekend.

From memory even when it was running, it had a bug preventing builds but I'm not sure.

**Comment by rmehner** [comment]
FYI: Homebrew recently switched to providing MySQL 8.+, which doesn't really work with the current Sequel Pro stable version (see #2699). Is there anything we can do to help with getting the release out? :)

**Comment by mwegner** [comment]
FYI on build stuff:  It may be easier to virtualize the build server.  ESXi has a free license that can be used for commercial work (just missing all of the cluster/etc enterprise features, but not very relevant here).  ESXi 6.5 update 2 is the best version for macOS VMs at the moment, since it can boot APFS volumes.

Virtualizing macOS is officially supported on Apple hardware, both by Apple and VMWare.  ESXi is basically a bare operating system that just runs VMs.  You could load it on a Mac Mini or older Mac Pro.

The interface to manage the VM is an HTML5 web client served off the ESXi host.  From the perspective of the VM, this *isn't* a remote desktop session.  You're seeing the display buffer directly, controlling virtual USB devices for input, etc.  So code signing password dialogs aren't an issue.  (I use macOS VMs in a few codesign-related projects).

It is possible to run macOS VMs on non-Apple hosts, but you need to modify ESXi to do this, since VMWare has some hardware ID checks in place to prevent it normally.

**Comment by jpike88** [comment]
Test builds are still broken.

**Comment by BookOfGreg** [comment]
@dmoagx My apologies for commenting off-topic in the other issue incorrectly.
May I request here that you fork and take over ownership of this project, so that you can release builds.
As proven by the summons at https://github.com/sequelpro/sequelpro/issues/1886#issuecomment-393491323 you are the only remaining active maintainer.

**Comment by dmoagx** [comment]
@BookOfGreg No problem, my comment wasn't directed at anyone in particular.
Forking (or rather - maintaining the forked stuff afterwards) is a quite involved process and my time and resources are rather limited (see my previous comments in this issue).
There are a lot of things to consider so this is an option I'm not too thrilled about.

**Comment by abhibeckert** [comment]
There’s no reason to fork. If somebody want so volounteer to do the work needed to do the next release, they can do so.

The main thing holding us back right now is some issues with our crash reporter and translation servers which should be fixed before doing a beta test period for the next release.

These two systems weren’t build by me and it’s taking me some time to figure out how they work and sort out the issues.

**Comment by Kaspik** [comment]
@abhibeckert Yo! Any news on your side? Anything how we can help with the crash reporter?

**Comment by chumbawumba** [comment]
@abhibeckert bumping again. Is there some amount of donations that we could reach that would help accelerate this process?

**Comment by jpike88** [comment]
@abhibeckert it would be even better if you could just do a 5 minute screencast of what the server situation looks like, and the nature of the issue. Mojave is about to be released and a whole bunch of people are about to have a very unpleasant surprise.

**Comment by jpike88** [comment]
@abhibeckert the above comment was made 20 days ago, and no response, yet you're actively committing to github. This project needs serious attention.

I've just emailed a request to join Slack with the goal of fixing the glaring Mojave bugs. But the condition is I get control of the repo and clear out the 900+ issues, placing the project in a state of LTS. If this fails I guess I'll just buy a copy of querious and call it a day

**Comment by dmoagx** [comment]
So here is my plan for the future:
Once Apple releases a new Mac Mini I'll purchase that and setup a new build server.
Since I'll have to use my own money for that and for legal reasons that will also be the point where I will fork.

**Comment by mobidev111** [comment]
No need to purchase a mac mini for building - there are plenty of cloud services which offer that for free to open source projects - either a barebone mac mini (e.g., https://www.macstadium.com/opensource) or build services (e.g. Visual Studio Team Services, https://azure.microsoft.com/en-us/free/open-source/)

**Comment by dmoagx** [comment]
The code signing key has no place in some cloud (aka "some else's computer").
Especially not, since Intel took the last year to thoroughly prove that they are incapable of engineering secure CPUs that will not leak data everywhere.

**Comment by mobidev111** [comment]
keeping the production signing key on a private machine is a valid requirement.

this however shouldn't stop people from revamping the build system for development & production builds - and then separately execute that logic for production on a local machine if that requirement still holds true.

**Comment by jpike88** [comment]
Is the problem thsst it wont build right now? Is tbe priority getting it to build or detangling the build server situation?

**Comment by dmoagx** [comment]
The reason because it doesn't build is (most likely) because the build script depends on resources from the tools server, which is also broken.
But, since the crash reports collector is/was also hosted on the tools server, it should IMHO be fixed first.

**Comment by Kaspik** [comment]
Yes, the server should be fixed first (as per 
https://github.com/sequelpro/sequelpro/issues/3034#issuecomment-406752647) and also because it's holding back daily builds on https://sequelpro.com/test-builds
I think most of us already sent request to join slack and help with project to @abhibeckert - but he is not responding, again.

**Comment by Kaspik** [comment]
Awww, great, @mattlangtree just invited me (thank you!) and I was able to join Slack. Let's see if we can move somewhere there, will keep you updated here as well.

**Comment by jpike88** [comment]
@mattlangtree could you invite me too? Would prefer Gitter though

**Comment by anthonysomerset** [comment]
I'm a systems engineer by trade rather than a developer or DBA. Do you need some expertise/assistance with fixing the servers if that hasn't already been offered by others?

**Comment by jpike88** [comment]
https://github.com/sequelpro/sequelpro/issues/3178#issuecomment-425025669

**Comment by Kaspik** [comment]
As @dmoagx mentioned in 
https://github.com/sequelpro/sequelpro/issues/3178#issuecomment-425025669 - things are moving now and we should know more after this weekend. They will hopefully find older backups, code that was running and we will try to migrate to some CI to fix servers and nightly builds.

**Comment by drew7721** [comment]
@mattlangtree send an invite my way please when you get a change. Thx.

**Comment by mattlangtree** [comment]
Anyone who is chasing acccess to slack, please send an email to contact@sequelpro.com. I’ll make sure you get added in.

**Comment by vinilanzoni** [comment]
Any updates on this thread?

**Comment by Kaspik** [comment]
Yes! It looks like crash reporter is back except translation server so we are trying to get the process working again without translating strings. (Will be english only for now)

**Comment by michaelkirkpatrick** [comment]
Hey team! Where do we stand today?

It's hard to keep track of what's being worked on from the public facing side, though I'm sure lots of great conversations are happening in Slack and I'm seeing commits happening. Is it worth updating the [Sequel 1.2 project board](https://github.com/sequelpro/sequelpro/projects/2) here in Github so folks can see what progress is being made and where things stand? If you need a PM type person to jump in and help out, I'm happy to lend a hand.

**Comment by jpike88** [comment]
I've asked the slack group to move to Gitter, and also asked that issues start getting culled. I don't have permissions to go beyond that unfortunately.

**Comment by Kaspik** [comment]
Release candidate is out! 🎉🚀 https://sequelpro.com/test-builds

**Comment by michaelkirkpatrick** [comment]
@Kaspik -- Great news! I'm not seeing anything on the releases page, will that be updated with the release candidate? https://github.com/sequelpro/sequelpro/releases

**Comment by Kaspik** [comment]
Not yet as it's not real release - but we are heading to that one soon! :)

**Comment by jpike88** [comment]
@Kaspik I think an issue cull is a good idea

**Comment by carkar7** [comment]
Will there be a fix to the slow query editor due to the touchbar?

**Comment by paulmillr** [comment]
Everything here is done except for some tiny feature: https://github.com/sequelpro/sequelpro/projects/2

#3116 is pretty important issue. 

@dmoagx @stuconnolly what's the deal here? Why can't you just push off a new release?

There is no excuse — instead of adding more and more new code, just focus on the freaking stable build. I maintain popular OSS software and know it's not a problem.

It has been more than two years now.

**Comment by Kaspik** [comment]
No one blocks you from using nigthly build.
We still need to fix the shortcut recorder and tabs in night mode first.

**Comment by paulmillr** [comment]
I've compiled it and using now. Someone who cannot compile software won't do that. So, it's pretty shitty solution.

**Comment by UnderpantsGnome** [comment]
@paulmillr https://sequelpro.com/test-builds

**Comment by dcposch** [comment]
@Kaspik respectfully, "fix the shortcut recorder and tabs in night mode first" is not good release management.

Sequel Pro has multiple commits in the last week (project is actually alive!), but has not been released in over two years (project appears dead).

It is a small, focused tool for connecting to MySQL from a Mac. That's all it does. Many people rely on it. And yet, it does not work at all with the **current stable release of MySQL**. That bug was actually fixed a year ago, and has still not been released.

Night Mode support is a nice-to-have. Fix it in a point release. You need to cut a 1.2 release now.

--

I realize it's dumb to just complain about a free open source project, so let me fix that.

I will donate $600 if you can commit to releasing 1.2 by the end of November. If you can commit to that, I'll donate $300 now and another $300 when it actually happens.

cc @dmoagx @stuconnolly @rowanbeentje

**Comment by dmoagx** [comment]
> I will donate $600 if you can commit to releasing 1.2 by the end of November. If you can commit to that, I'll donate $300 now and another $300 when it actually happens.

Aaaaaand again, for the umptieth time: Money is not an issue for the project - Time is! 
Also it is scientifically proven, that money as an incentive does not work.
If you want to spend that amount of money, personally talk to one of the developers if you could pay them by the hour.

**Comment by michaelkirkpatrick** [comment]
Thanks for all your work on this team! And from the United States, Happy Thanksgiving!

**Comment by michaelkirkpatrick** [comment]
Hey team, checking in. What's the latest on moving from test builds to a release candidate and then to an official release?

**Comment by paulmillr** [comment]
@michaelkirkpatrick Simple. There is no plan — maintainers don't care about such mundane tasks.

**Comment by dmoagx** [comment]
@michaelkirkpatrick 
See https://github.com/sequelpro/sequelpro/projects/2

On an unrelated note: I am the wrong person to ask about doing a release.

**Comment by paulmillr** [comment]
@dmoagx those "issues" are definitely not blockers.

**Comment by anotherdevs** [comment]
@dmoagx Is the plan still to fork this project? Is the basic edition of the Mac Mini sufficient to run it? https://www.apple.com/de/shop/buy-mac/mac-mini/3,6-ghz-quad%E2%80%91core-prozessor-128gb#

**Comment by jpike88** [comment]
It is pretty amazing that a proper release hasn't been organised, there are a couple of serious bug in Mojave that non beta channel users have to contend with.

**Comment by dmoagx** [comment]
@anotherdevs Yes, as things stand my plan is still to fork, but since I'm pretty busy myself I'm not sure how sustainable that effort would be…

**Comment by origamifreak2** [comment]
> No one blocks you from using nigthly build.
> We still need to fix the shortcut recorder and tabs in night mode first.

This might be a dumb question, but how stable are the test builds in general? Is there a particular test build that would be a good stop gap measure until a new release? I think most users might just want the app to stop crashing when closing tabs, but I personally am a little nervous to use the test builds on actual data unless I know they're pretty stable.

Thanks

**Comment by jfi** [comment]
I use them in daily use across a large number of different versions of
MySQL/MariaDB and haven’t found any major issues since they started being
rebuilt nightly. YMMV of course but they seem, functionally, generally very
stable. The filter bar disappearing is a very annoying change, but it’s
good to see that on the roadmap for a fix.

**Comment by Kaspik** [comment]
@origamifreak2 I use them daily as well.

**Comment by mkismy** [comment]
I understand things aren't that simple, but could you release a single 1.1.3 patch that will fix #3116 ?

**Comment by BookOfGreg** [comment]
This project is the 2nd post on Hacker News today:
https://news.ycombinator.com/item?id=18936509
Plenty folk saying they love this project there, looking forward to when we're able to release again.

**Comment by stychos** [comment]
What is needed to reanimate the project?

**Comment by rowanbeentje** [comment]
I think 1.2 is waiting on the "required" issues on https://github.com/sequelpro/sequelpro/projects/2 .

**Comment by stychos** [comment]
I understand :-) Maybe it's a time to release something like 1.1.3 reducing count of blocking (from the point of regular users) troubles? Now with the latest @dmoagx and other people commits most annoying troubles are resolved: it's pretty stable on Mojave and looks not very bad in the dark mode. That will be enough for the most of "hot" issues.

But my question is about other. Maybe I can help in some way? Unfortunately I am not Objective C or macOS developer, but I can try to help if I will get some instructions and points to look for. Also if there's troubles with the nightly/testing builds I can share my hardware if it's possible to do remotely via command line.

**Comment by origamifreak2** [comment]
Just wanted to put in my two cents. I've been using the test builds for the past little while, and I haven't noticed any errors. I'm still a little nervous to use it on live data, so I've mainly been using MySQL Workbench for that. I realize the developers have limited time, but I'm sure a release which fixes the major crash causing issues would be greatly appreciated by those like me who aren't completely comfortable with the test builds.

**Comment by Kaspik** [comment]
Conversation moved to newer thread https://github.com/sequelpro/sequelpro/issues/3385 - closing this one as duplicate and outdated.
